charts/team-ns/templates/argocd/argocd-project.yaml

```diff
@@ -0,0 +1,81 @@
+{{- $v := .Values }}
+{{- $a := $v.apps.argocd }}
+{{- if and (not (eq $v.teamId "admin")) $a.enabled }}
+apiVersion: argoproj.io/v1alpha1
+kind: AppProject
+metadata:
+  name: team-{{ $v.teamId }}
+  namespace: argocd
+  # Finalizer that ensures that project is not deleted until it is not referenced by any application
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  description: ArgoCD project for team {{ $v.teamId }}
+  # Allow manifests to deploy from any Git repos
+  sourceRepos:
+  - https://gitea.{{ $v.cluster.domainSuffix }}/otomi/team-{{ $v.teamId }}-argocd.git
+  # Only permit applications to deploy to the guestbook namespace in the same cluster
+  destinations:
+  - namespace: team-{{ $v.teamId }}
+    server: https://kubernetes.default.svc
+  # Deny all cluster-scoped resources from being created
+  clusterResourceWhitelist: []
+  # - group: ''
+  #   kind: Namespace
+  # Allow all namespaced-scoped resources to be created, except for ResourceQuota, LimitRange, NetworkPolicy
+  namespaceResourceBlacklist:
+  - group: ''
+    kind: ResourceQuota
+  - group: ''
+    kind: LimitRange
+  - group: ''
+    kind: NetworkPolicy
+  # Deny all namespaced-scoped resources from being created
+  # namespaceResourceWhitelist: []
+  # - group: 'apps'
+  #   kind: Deployment
+  # - group: 'apps'
+  #   kind: StatefulSet
+  roles:
+{{- if $v.otomi.isMultitenant }}
+  # we create a scoped team-admin role since we are only allowed access to team-* projects as team-admin in multitenant setup
+  - name: team-admin
+    description: Team member privileges to team-{{ $v.teamId }}
+    policies:
+    - p, proj:team-{{ $v.teamId }}:team-admin, *, *, team-{{ $v.teamId }}/*, allow
```

Comment:
I do not understand the meaning of each field here.
Could you add the comment that explains their meaning or provide a link to the documentation?

Reply:
the docs are easy to find...we've had this discussion before

Reply:
you could save a lot of my time by simply describing the syntax in the comment